src/main/java/org/apache/maven/plugins/changes/model/AbstractAction.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.maven.plugins.changes.model;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Abstract calls with helper methods for {@link Action}
+ */
+public abstract class AbstractAction {
+
+    private List<DueTo> dueTosList;
+
+    private List<String> fixedIssueList;
+
+    public abstract String getDueTo();
+
+    public abstract String getDueToEmail();
+
+    public abstract String getFixedIssuesString();
+
+    /**
+     * Parse due-to and due-to-email attributes
+     *
+     * @return a List of due-to person
+     */
+    public List<DueTo> getDueTos() {
+        if (dueTosList != null) {
+            return dueTosList;
+        }
+        dueTosList = new ArrayList<>();
```

Comment:
This doesn't feel thread-safe. Not sure how much that matters. It's hard to analyze because of the split between the superclass and the concrete subclass. See https://stackoverflow.com/questions/8297705/how-to-implement-thread-safe-lazy-initialization

Reply:
Whole model is loaded once and use as local variable - please see usage of `ChangesXML`

Reply:
could this all be done upfront in the constructor then?

Reply:
no - fields are populated by xml reader after class creation